protocol_rfcs/collated-string-type.md

```diff
@@ -0,0 +1,146 @@
+# Collated String Type
+**Associated Github issue for discussions: https://github.com/delta-io/delta/issues/2894**
+
+This protocol change adds support for collated strings.
+
+--------
+
+> ***Add a new section in front of the [Primitive Types](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#primitive-types) section.***
+
+### Collations
+
+Collations are a set of rules for how strings are compared. They do not affect how strings are stored. Collations are applied when comparing strings for equality or to determine the sort order of two strings. Case insensitive comparison is one example of a collation where case is ignored when string are compared for equality and the lower cased variant of a string is used to determine its sort order.
+
+Collations can be specified for all string fields in a table schema. It is also possible to store statistics per collation version. This is required because the min and max values of a column can differ based on the used collation or collation version.
+
+By default, all strings are collated using binary collation. That means that strings compare equal if their binary representations are equal. The binary representation is also used to sort them.
+
+#### Collation identifiers
+
+Collations can be referred to using collation identifiers. The Delta format does not specify any collation rules other than binary collation, but supports the concept of collation providers such that engines can use providers like [ICU](https://icu.unicode.org/) and mark statistics accordingly.
+
+A collation identifier consists of 3 parts, which are combined into one identifier using dots as separators. Dots are not allowed to be part of provider and collation names, but can be used in versions.
+
+Part | Description
+-|-
+Provider | Name of the provider. Must not contain dots
+Name | Name of the collation as provided by the provider. Must not contain dots
+Version | Version string. Is allowed to contain dots
+
+#### Specifying collations in the table schema
+
+Collations can be specified for all string types in a schema. This includes string fields, but also the key and value type of maps and the element type of arrays. Collations are specified in the metadata of the closest enclosing field.
+Collation identifiers are stored in a `collations` object. These object can have 4 keys:
+
+Key | Value | Description
+-|-|-
+collation | collation identifier | Collation of a string field. Only valid when none of the other keys are present.
+elementCollation | `collations object` | Collations of elements in an array. Only valid when none of the other keys are present.
```

Comment:
this is not the case since we moved collation to the metadata and not an additional field, same for below